src/cirq_iqm/iqm_remote.py

```diff
@@ -0,0 +1,129 @@
+# Copyright 2020–2021 Cirq on IQM developers
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""
+Implements a CIRQ compatible sampler that calls the IQM backend
+"""
+import os
+import cirq
+import numpy as np
+from cirq import study
+from cirq.study import resolver
+from cirq_iqm.iqm_client import IQMBackendClient, IQMCircuit, IQMInstruction
+
+
+def get_sampler_from_env() -> 'IQMSampler':
+    """
+    Initialize an IQM sampler using environment variables IQM_TOKEN and IQM_URL
+
+    Returns:
+        IQM Sampler
+    """
+    iqm_url = os.environ.get('IQM_URL')
+    iqm_token = os.environ.get('IQM_TOKEN')
+    if not iqm_url or not iqm_token:
+        raise EnvironmentError('Environment variables IQM_URL and IQM_TOKEN are not set.')
+    return IQMSampler(url=iqm_url, token=iqm_token)
+
+
+def _serialize_iqm(circuit: cirq.Circuit) -> IQMCircuit:
+    """
+    Converts cirq circuit to IQM compatible representation.
+    Args:
+        circuit: Circuit to serialize
+
+    Returns:
+        IQM circuit object
+    """
+    instructions = []
+    for moment in circuit.moments:
+        for operation in moment.operations:
+            gate_dict = operation.gate._json_dict_()
+            instructions.append(
+                IQMInstruction(
+                    name=gate_dict['cirq_type'],
+                    qubits=[str(qubit) for qubit in operation.qubits],
+                    args={key: val for key, val in gate_dict.items() if key != 'cirq_type'}
+                )
+            )
+
+    circuit_dict = IQMCircuit(
+        name='Serialized from cirq',
+        instructions=instructions,
+        args={}  # todo: implement arguments
+    )
+    return circuit_dict
+
+
+class IQMSampler(cirq.work.Sampler):
+    """
+    IQM implementation of a cirq sampler.
+    Allows to sample circuits using a real backend
+    """
+
+    def __init__(self, url, token):
+        self._client = IQMBackendClient(url, token)
+
+    def run_sweep(
```

Comment:
We don't really support sweep, so should this just throw an NotImplementedException at all times? It's potentially a bit confusing. At the very least, the docs should not imply that we support sweeps :)

Reply:
This is needed for compatibility with IQM. I've added this to the docstring for clarity